functions/tests/actions/wayback-machine.spec.js

```diff
@@ -0,0 +1,64 @@
+const axios = require('axios');
+const {expect} = require('chai');
+const rewire = require('rewire');
+
+const action = rewire('../../src/actions/wayback-machine');
+
+const mockApp = require('../_utils/mocking/platforms/app');
+const mockDialog = require('../_utils/mocking/dialog');
+
+describe('actions', () => {
+  describe('wayback machine', () => {
+    let app;
+    let dialog;
+    let archiveRequest = axios.get('http://web.archive.org/__wb/search/metadata?q=cnn.com');
+    let alexaRequest = axios.get('http://data.alexa.com/data?cli=10&url=cnn.com');
+
+    beforeEach(() => {
+      app = mockApp();
+      dialog = mockDialog();
+      action.__set__('dialog', dialog);
+    });
+
+    it('check to see that overall action eventually returns a promise', () => {
+      action.handler(app);
+      expect(Promise.resolve()).to.be.a('promise');
+    });
+
+    it('check to see that axios request promises are working', () => {
+      expect(archiveRequest).to.not.be.undefined;
+      expect(alexaRequest).to.not.be.undefined;
+    });
+
+    it('check to see that archiveEngine is working', () => {
+      let obj = { earliestYear: 0, latestYear: 0, totalUniqueURLs: 0 };
+      let result;
+
+      action.__set__('archiveEngine', function (a, b) {
```

Comment:
it seems this test has never come inside of this mock. You never call `handler` in this test on others. Actually, it is not clear what: `check to see that archiveEngine is working` means here, could you be more specific?